### PR TITLE
[WIP] Port optimized yarn install method from reaction

### PR DESCRIPTION
**Pull request by zephraph**
The yarn install step for force can take around a minute and a half to finish. There's no caching happening currently. This is an attempt to mitigate that by adding the same cache method that reaction uses.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
+  save_cache:
+    key: yarn-deps-{{ checksum "yarn.lock" }}
+    paths:
+      - ./node_modules
+
+restore_cache: &restore_cache
+  restore_cache:
+    keys:
+    - yarn-deps-{{ checksum "yarn.lock" }}
+    - yarn-deps-
```
Comment (alloy) [inline .circleci/config.yml]: What is this second key about?

Comment (zephraph) [reply .circleci/config.yml]: It's a fallback. CircleCI has an immutable caching structure. It saves a cache with an associated key and will never change that version of the cache after the first write. When reading from the cache, it _always_ does a partial string match. If it doesn't find a match to the first key, it'll fall over to the next. That means in this case it'll load the last version of the cache.

Comment (damassi) [reply .circleci/config.yml]: Nice

File: yarn.lock [hunk 2]
```diff
@@ -1061,6 +1061,7 @@ antigravity@artsy/antigravity:
   version "0.1.3"
   resolved "https://codeload.github.com/artsy/antigravity/tar.gz/a4438d2fe9d0cdf71f1c47faba371cd3d004e140"
   dependencies:
+    coffeescript "1.11.1"
```
Comment (joeyAghion) [inline yarn.lock]: 🤔

File: Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
-FROM bitnami/node:8
-#FROM circleci/node:8-stretch-browsers
+# FROM bitnami/node:8
+FROM circleci/node:8
```
Comment (joeyAghion) [inline Dockerfile]: This is the Dockerfile used to actually build and run the production app, right? Isn't it strange to base it on Circle's, which presumably includes a lot of stuff that's unnecessary in prod?

Comment (zephraph) [reply Dockerfile]: Yep, it was a failed experiment. I didn't understand what the hokusai setup was doing. @ashfurrow helped me understand it a bit better. Ultimately I think I'll need to learn a little more before I can take this anywhere. 

Right now it _seems_ impossible to use circleci's caching for the test step (which is by far the longest). It's also going through the process of downloading yarn every time. I'll be reverting this change, but I'm curious if we can take the test step off of the hokusai image.... if not that, I think optimizing the force build is going to be a really, really hard process.

File: yarn.lock (hunk 2 above)
Comment (orta) [reply yarn.lock]: Yeah, older artsy dependencies shipped coffescript code, not compiled javascript - I've updated one of two of these, but not all

**Pull request closed without merging**